crates/sui-framework/packages/move-stdlib/sources/ascii.move

```diff
@@ -27,82 +28,139 @@ module std::ascii {
     }
 
+    /// Append the `other` string to the end of `string`.
+    public fun append(string: &mut String, other: String) {
+        string.bytes.append(other.into_bytes())
+    }
+
+    /// Insert the `other` string at the `at` index of `string`.
+    public fun insert(s: &mut String, at: u64, o: String) {
+        assert!(at <= s.length(), EInvalidIndex);
+        o.into_bytes().destroy!(|e| s.bytes.insert(e, at));
+    }
+
+    /// Copy the slice of the `string` from `i` to `j` into a new `String`.
+    public fun substring(string: &String, i: u64, j: u64): String {
+        assert!(i <= j && j <= string.length(), EInvalidIndex);
+        let mut bytes = vector[];
+        i.range_do!(j, |i| bytes.push_back(string.bytes[i]));
+        String { bytes }
+    }
+
     /// Get the inner bytes of the `string` as a reference
     public fun as_bytes(string: &String): &vector<u8> {
-       &string.bytes
+        &string.bytes
     }
 
     /// Unpack the `string` to get its backing bytes
     public fun into_bytes(string: String): vector<u8> {
-       let String { bytes } = string;
-       bytes
+        let String { bytes } = string;
+        bytes
     }
 
-    /// Unpack the `char` into its underlying byte.
+    /// Unpack the `char` into its underlying bytes.
     public fun byte(char: Char): u8 {
-       let Char { byte } = char;
-       byte
+        let Char { byte } = char;
+        byte
     }
 
-    /// Returns `true` if `b` is a valid ASCII character. Returns `false` otherwise.
+    /// Returns `true` if `b` is a valid ASCII character.
+    /// Returns `false` otherwise.
     public fun is_valid_char(b: u8): bool {
-       b <= 0x7F
+        b <= 0x7F
     }
 
-    /// Returns `true` if `byte` is an printable ASCII character. Returns `false` otherwise.
+    /// Returns `true` if `byte` is an printable ASCII character.
+    /// Returns `false` otherwise.
     public fun is_printable_char(byte: u8): bool {
-       byte >= 0x20 && // Disallow metacharacters
-       byte <= 0x7E // Don't allow DEL metacharacter
+        byte >= 0x20 && // Disallow metacharacters
+        byte <= 0x7E // Don't allow DEL metacharacter
+    }
+
+    /// Returns `true` if `string` is empty.
+    public fun is_empty(string: &String): bool {
+        string.bytes.is_empty()
+    }
+
+    /// Convert a `string` to its uppercase equivalent.
+    public fun to_uppercase(string: &String): String {
+        let bytes = string.as_bytes().map_ref!(|byte| char_to_uppercase(*byte));
+        String { bytes }
+    }
+
+    /// Convert a `string` to its lowercase equivalent.
+    public fun to_lowercase(string: &String): String {
+        let bytes = string.as_bytes().map_ref!(|byte| char_to_lowercase(*byte));
+        String { bytes }
+    }
+
+    /// Computes the index of the first occurrence of the `substr` in the `string`.
+    /// Returns the length of the `string` if the `substr` is not found.
+    /// Returns 0 if the `substr` is empty.
```

Comment:
I really hate this API, but I agree it is the right thing to do for consistency

Reply:
Same here, not a fan.